File: web/src/lib/stark/repository/DataRepository.ts
import type {
  Account,
  Asset,
  Budget,
  CategoryRule,
  ExchangeRate,
  ImportErrorLog,
  ImportResult,
  Loan,
  SavingsGoal,
  SavingsPlan,
  ThemeConfig,
  Transaction,
  User,
} from "@/lib/stark/models/types";

export interface DataRepository {
  getCurrentUser(): Promise<User | null>;
  saveUser(user: User): Promise<void>;

  getAccounts(): Promise<Account[]>;
  getAccount(id: string): Promise<Account | null>;
  saveAccount(account: Account): Promise<void>;
  deleteAccount(id: string): Promise<void>;

  getTransactions(accountId: string, page?: number, pageSize?: number): Promise<Transaction[]>;
  getTransaction(id: string): Promise<Transaction | null>;
  saveTransaction(transaction: Transaction): Promise<void>;
  deleteTransaction(id: string): Promise<void>;
  importTransactions(transactions: Transaction[]): Promise<ImportResult>;

  getAssets(accountId: string): Promise<Asset[]>;
  saveAsset(asset: Asset): Promise<void>;
  deleteAsset(id: string): Promise<void>;

  getBudgets(accountId: string): Promise<Budget[]>;
  saveBudget(budget: Budget): Promise<void>;
  deleteBudget(id: string): Promise<void>;

  getLoans(accountId: string): Promise<Loan[]>;
  saveLoan(loan: Loan): Promise<void>;
  deleteLoan(id: string): Promise<void>;

  getSavingsGoals(accountId: string): Promise<SavingsGoal[]>;
  saveSavingsGoal(goal: SavingsGoal): Promise<void>;
  deleteSavingsGoal(id: string): Promise<void>;
  getSavingsPlans(goalId: string): Promise<SavingsPlan[]>;
  saveSavingsPlan(plan: SavingsPlan): Promise<void>;

  getCategoryRules(accountId: string): Promise<CategoryRule[]>;
  saveCategoryRule(rule: CategoryRule): Promise<void>;

  getImportErrorLogs(accountId: string): Promise<ImportErrorLog[]>;
  saveImportErrorLog(log: ImportErrorLog): Promise<void>;

  getExchangeRates(): Promise<ExchangeRate[]>;

  getThemeConfig(userId: string): Promise<ThemeConfig | null>;
  saveThemeConfig(config: ThemeConfig): Promise<void>;
}

File: composeApp/src/main/java/com/wotty/stark/data/repository/DataRepository.kt
package com.wotty.stark.data.repository

import com.wotty.stark.data.model.*

/**
 * 统一数据仓库接口 - 所有 ViewModel 只依赖此接口
 */
interface DataRepository {
    suspend fun getCurrentUser(): User?
    suspend fun saveUser(user: User)

    suspend fun getAccounts(): List<Account>
    suspend fun getAccount(id: String): Account?
    suspend fun saveAccount(account: Account)
    suspend fun deleteAccount(id: String)

    suspend fun getTransactions(accountId: String, page: Int = 1, pageSize: Int = 50): List<Transaction>
    suspend fun getTransaction(id: String): Transaction?
    suspend fun saveTransaction(transaction: Transaction)
    suspend fun deleteTransaction(id: String)
    suspend fun importTransactions(transactions: List<Transaction>): ImportResult

    suspend fun getAssets(accountId: String): List<Asset>
    suspend fun saveAsset(asset: Asset)
    suspend fun deleteAsset(id: String)

    suspend fun getBudgets(accountId: String): List<Budget>
    suspend fun saveBudget(budget: Budget)
    suspend fun deleteBudget(id: String)

    suspend fun getLoans(accountId: String): List<Loan>
    suspend fun saveLoan(loan: Loan)
    suspend fun deleteLoan(id: String)

    suspend fun getSavingsGoals(accountId: String): List<SavingsGoal>
    suspend fun saveSavingsGoal(goal: SavingsGoal)
    suspend fun deleteSavingsGoal(id: String)
    suspend fun getSavingsPlans(goalId: String): List<SavingsPlan>
    suspend fun saveSavingsPlan(plan: SavingsPlan)

    suspend fun getCategoryRules(accountId: String): List<CategoryRule>
    suspend fun saveCategoryRule(rule: CategoryRule)

    suspend fun getImportErrorLogs(accountId: String): List<ImportErrorLog>
    suspend fun saveImportErrorLog(log: ImportErrorLog)

    suspend fun getExchangeRates(): List<ExchangeRate>

    suspend fun getThemeConfig(userId: String): ThemeConfig?
    suspend fun saveThemeConfig(config: ThemeConfig)

    /** 首次使用无数据时，种入一套当月演示数据 */
    suspend fun seedDemoData()
}

data class ImportResult(
    val imported: Int = 0,
    val skipped: Int = 0,
    val errors: Int = 0
)
